src/ophys_etl/modules/downsample_video/utils.py

```diff
@@ -0,0 +1,637 @@
+                                    border:video_shape[1]-border,
+                                    border:video_shape[2]-border]
+        if quantiles is not None:
+            q0, q1 = np.quantile(full_data, quantiles)
+        else:
+            q0 = full_data.min()
+            q1 = full_data.max()
+        print('got normalization')
+
+    return q0, q1
+
+
+def _video_array_from_h5(
+        h5_path: pathlib.Path,
+        min_val: float,
+        max_val: float,
+        reticle: bool = True,
+        d_reticle: int = 64) -> np.ndarray:
+    """
+    Read in an HDF5 file and convert it into a numpy array that
+    can be passed to _write_array_to_video
+
+    Parameters
+    ----------
+    h5_path: pathlib.Path
+        Path to the HDF5 file containing the video data
+
+    min_val: float
+        The minimum value at which to clip the video
+
+    max_val: float
+        The maximum value at which to clip the video
+
+    reticle: bool
+        If True, add a grid of red lines to the video, to help guide
+        the eye
+
+    d_reticle: int
+        Spacing between reticles
+
+    Returns
+    -------
+    video_as_uint: np.ndarray
+        A (ntime, nrows, ncols, 3) array of uints representing the
+        RGB video.
+    """
+
+    with h5py.File(h5_path, 'r') as in_file:
+        video_shape = in_file['data'].shape
+
+        video_as_uint = np.zeros((video_shape[0],
+                                  video_shape[1],
+                                  video_shape[2],
+                                  3), dtype=np.uint8)
+        dt = 500
+        for i0 in range(0, video_shape[0], dt):
+            i1 = i0+dt
+            data = in_file['data'][i0:i1, :, :].astype(float)
+            data = np.where(data > min_val, data, min_val)
+            data = np.where(data < max_val, data, max_val)
+            delta = max_val-min_val
+            data = np.round(255.0*(data-min_val)/delta).astype(np.uint8)
+            for ic in range(3):
+                video_as_uint[i0:i1, :, :, ic] = data
+
+    print('constructed video_as_uint')
+
+    if reticle:
+        for ii in range(d_reticle, video_shape[1], d_reticle):
+            old_vals = np.copy(video_as_uint[:, ii:ii+2, :, :])
+            new_vals = np.zeros(old_vals.shape, dtype=np.uint8)
+            new_vals[:, :, :, 0] = 255
+            new_vals = (new_vals//2) + (old_vals//2)
+            new_vals = new_vals.astype(np.uint8)
+            video_as_uint[:, ii:ii+2, :, :] = new_vals
+        for ii in range(d_reticle, video_shape[2], d_reticle):
+            old_vals = np.copy(video_as_uint[:, :, ii:ii+2, :])
+            new_vals = np.zeros(old_vals.shape, dtype=np.uint8)
+            new_vals[:, :, :, 0] = 255
+            new_vals = (new_vals//2) + (old_vals//2)
+            new_vals = new_vals.astype(np.uint8)
+            video_as_uint[:, :, ii:ii+2, :] = new_vals
+
+    print('added reticles')
+
+    return video_as_uint
+
+
+def create_downsampled_video(
```

Comment:
Can you please move this function  to the top of the module since it is the most user-facing function in this module and I would want to see it first when I open the module?